src/platform/Linux/ConnectivityManagerImpl.cpp

```diff
@@ -2301,37 +2297,30 @@ void ConnectivityManagerImpl::_OnWpaInterfaceScanDone(WpaSupplicant1Interface *
     }
 
     std::vector<WiFiScanResponse> * networkScanned = new std::vector<WiFiScanResponse>();
-    for (const gchar * bssPath = (bsss != nullptr ? *bsss : nullptr); bssPath != nullptr; bssPath = *(++bsss))
+    for (const char * bssPath = (bsss != nullptr ? *bsss : nullptr); bssPath != nullptr; bssPath = *(++bsss))
     {
         WiFiScanResponse network;
         if (_GetBssInfo(bssPath, network))
         {
-            if (sInterestedSSIDLen == 0)
-            {
-                networkScanned->push_back(network);
-            }
-            else if (network.ssidLen == sInterestedSSIDLen && memcmp(network.ssid, sInterestedSSID, sInterestedSSIDLen) == 0)
+            if (sInterestedSSIDLen == 0 || memcmp(network.ssid, sInterestedSSID, sInterestedSSIDLen) == 0)
```

Comment:
I think that the check checking the length `network.ssidLen == sInterestedSSIDLen` should stay. In absence of it I am afraid that the memcmp will succeed even in cases where sInterestedSSID will be just a prefix of network.ssid. For example:

`sInterestedSSID` == `TEST` (`sInterestedSSIDLen` == 4) will match `network.ssid` == `TEST123`

Reply:
Many thanks for spotting that! Fixed.